content/docs/components-and-props.md

```diff
@@ -16,57 +16,57 @@ prev: rendering-elements.html
+Les composants vous permettent de diviser l'interface utilisateur en éléments indépendants et réutilisables, permettant ainsi de considérer chaque élément de manière isolée. Cette page fournit une introduction au concept de composant. Vous pouvez trouver une [référence d'API de composant détaillée ic](/docs/react-component.html).
 
-Conceptually, components are like JavaScript functions. They accept arbitrary inputs (called "props") and return React elements describing what should appear on the screen.
+Conceptuellement, les composants sont comme des fonctions JavaScript. Ils acceptent des entrées arbitraires (appelées "props") et renvoient des éléments React décrivant ce qui doit apparaître à l'écran.
 
-## Function and Class Components {#function-and-class-components}
+## Fonctions composants et composants à base de classe {#function-and-class-components}
 
-The simplest way to define a component is to write a JavaScript function:
+Le moyen le plus simple de définir un composant consiste à écrire une fonction JavaScript:
 
 ```js
 function Welcome(props) {
-  return <h1>Hello, {props.name}</h1>;
+  return <h1>Bonjour, {props.name}</h1>;
 }
 ```
 
-This function is a valid React component because it accepts a single "props" (which stands for properties) object argument with data and returns a React element. We call such components "function components" because they are literally JavaScript functions.
+Cette fonction est un composant React valide car elle accepte un seul argument "props" (qui se traduit par "propriétés") contenant des données et renvoie un élément React. Nous appelons de tels composants "fonctions composants" car sont littéralement des fonctions JavaScript.
 
-You can also use an [ES6 class](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes) to define a component:
+Vous pouvez également utiliser une [classe ES6](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Classes) pour définir un composant:
 
 ```js
 class Welcome extends React.Component {
   render() {
-    return <h1>Hello, {this.props.name}</h1>;
+    return <h1>Bonjour, {this.props.name}</h1>;
   }
 }
 ```
 
-The above two components are equivalent from React's point of view.
+Les deux composants ci-dessus sont équivalents d'un point de vue React.
 
-Classes have some additional features that we will discuss in the [next sections](/docs/state-and-lifecycle.html). Until then, we will use function components for their conciseness.
+Les classes possèdent quelques fonctionnalités supplémentaires dont nous discuterons dans les [sections suivantes](/docs/state-and-lifecycle.html). En attendant, nous utiliserons les fonctions composants pour leur forme concise.
 
-## Rendering a Component {#rendering-a-component}
+## Rendu d'un Component {#rendering-a-component}
 
-Previously, we only encountered React elements that represent DOM tags:
+Jusque là, nous avons uniquement rencontré des éléments React sous forme de balise DOM:
```

Comment:
Maybe use "Jusqu'içi" instead of "jusque là", the first one refer to 'now' and the second to a known moment, both are quite right in this sentence, but "jusqu'içi" may be more accurate

Reply:
C'est juste, sauf qu'il n'y a pas de cédille : « jusqu’ici »